catalog/core/catalog-core-impl/filter-proxy/src/main/java/ddf/catalog/filter/proxy/adapter/GeotoolsFilterAdapterImpl.java

```diff
@@ -130,42 +129,48 @@ public Object visit(Divide expression, Object delegate) {
         throw new UnsupportedOperationException(Divide.NAME + " expression not supported.");
     }
 
+    @Override
     public Object visit(Function expression, Object delegate) {
         if (expression == null) {
             throw new UnsupportedOperationException(
                     Function.class.getSimpleName() + " must not be null.");
         }
+        List<Expression> parameters = expression.getParameters();
 
-        if (expression.getName()
-                .equals(ProximityFunction.NAME.getName())) {
-            List<Expression> parameters = new ArrayList<>(expression.getParameters());
-            if (parameters.size() == ProximityFunction.NUM_PARAMETERS) {
-
-                String propertyName = parameters.remove(0)
-                        .toString();
-                int distance;
-                try {
-                    distance = Integer.parseInt(parameters.remove(0)
-                            .toString());
-                } catch (NumberFormatException e) {
-                    throw new IllegalArgumentException("Distance parameter was not valid");
-                }
-                String searchTerm = parameters.remove(0)
-                        .toString();
-                return new ProximityArguments(propertyName, distance, searchTerm);
-            } else {
-                throw new IllegalArgumentException("Not enough arguments for ProximityFunction");
-            }
+        int argCount = expression.getFunctionName()
+                .getArgumentCount();
+
+        if (parameters == null || parameters.size() != argCount) {
+            throw new UnsupportedOperationException(
+                    Function.class.getSimpleName() + " requires " + argCount + " arguments.");
+        }
+        //unwrap the function arguments and return them
+        List<Object> ret = new ArrayList<>(argCount);
```

Comment:
✏️ There's usually no reason to pre-size an `ArrayList`, particularly when it is going to be short, but it obviously doesn't hurt to do so.